### ISSUE TITLE
Hash Sum mismatch

### ISSUE THREAD

**Issue by tuanbs** [pculture/unisubs#2446]
Hi, I followed the "Quick Start" of this project but at the step 3 (Build the Amara docker image) I got this error:
"...
Get:13 http://archive.ubuntu.com trusty/universe amd64 Packages [7589 kB]
Fetched 20.8 MB in 53s (391 kB/s)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/main/binary-amd64/Packages  Hash Sum mismatch

E: Some index files failed to download. They have been ignored, or old ones used instead.
The command '/bin/sh -c apt-get update' returned a non-zero code: 100
"

- Then I ignored it and went to step 4 and got this error:
"
unisubs_cache_1 is up-to-date
unisubs_queue_1 is up-to-date
unisubs_db_1 is up-to-date
Pulling worker (amara-app:latest)...
Pulling repository docker.io/library/amara-app
ERROR: Error: image library/amara-app:latest not found
"
- Keep ignoring and went to step 5 and got this error:
"
Pulling app (amara-app:latest)...
Pulling repository docker.io/library/amara-app
ERROR: Error: image library/amara-app:latest not found
Pulling app (amara-app:latest)...
Pulling repository docker.io/library/amara-app
ERROR: Error: image library/amara-app:latest not found
Pulling app (amara-app:latest)...
Pulling repository docker.io/library/amara-app
ERROR: Error: image library/amara-app:latest not found
Pulling app (amara-app:latest)...
Pulling repository docker.io/library/amara-app
ERROR: Error: image library/amara-app:latest not found
"

- At step 6 I cannot access the site at <http://unisubs.example.com:8000>.

Please tell me how I can fix it. Thank you.

**Comment by bendk** [comment]
I've noticed this recently as well on docker builds.  What if you put this line before the one that fails:
```
RUN apt-get update
```

**Comment by tuanbs** [comment]
I opened "Docker Quickstart Terminal" -> cd to the unisubs project -> typed: RUN apt-get update -> it showed: "bash: RUN: command not found".
Then I did step 3 again: bin/dev build -> still the same error below:
"
Sending build context to Docker daemon 404.9 MB
Step 1 : FROM ubuntu:14.04
 ---> c4bea91afef3
Step 2 : MAINTAINER Amara "http://amara.org"
 ---> Running in 4240f6946972
 ---> 3e072c02e9eb
Removing intermediate container 4240f6946972
Step 3 : RUN (echo "deb http://archive.ubuntu.com/ubuntu trusty main universe multiverse" > /etc/apt/sources.list)
 ---> Running in ceadd7aa1a18
 ---> b7a10721f42d
Removing intermediate container ceadd7aa1a18
Step 4 : RUN (echo "deb-src http://archive.ubuntu.com/ubuntu trusty main universe multiverse" >> /etc/apt/sources.list)
 ---> Running in 415255076b46
 ---> 519837f1cfce
Removing intermediate container 415255076b46
Step 5 : RUN (echo "deb http://archive.ubuntu.com/ubuntu trusty-updates main universe multiverse" >> /etc/apt/sources.list)
 ---> Running in 0366fda6fa3f
 ---> 571c36682b10
Removing intermediate container 0366fda6fa3f
Step 6 : RUN (echo "deb-src http://archive.ubuntu.com/ubuntu trusty-updates main universe multiverse" >> /etc/apt/sources.list)
 ---> Running in b6805870649f
 ---> c493a45d9c7e
Removing intermediate container b6805870649f
Step 7 : RUN apt-get update
 ---> Running in a95312eb5437
Ign http://archive.ubuntu.com trusty InRelease
Get:1 http://archive.ubuntu.com trusty-updates InRelease [64.4 kB]
Hit http://archive.ubuntu.com trusty Release.gpg
Get:2 http://archive.ubuntu.com trusty-updates/main Sources [311 kB]
Get:3 http://archive.ubuntu.com trusty-updates/universe Sources [183 kB]
Get:4 http://archive.ubuntu.com trusty-updates/multiverse Sources [5116 B]
Get:5 http://archive.ubuntu.com trusty-updates/main amd64 Packages [836 kB]
Get:6 http://archive.ubuntu.com trusty-updates/universe amd64 Packages [432 kB]
Get:7 http://archive.ubuntu.com trusty-updates/multiverse amd64 Packages [14.1 kB]
Hit http://archive.ubuntu.com trusty Release
Get:8 http://archive.ubuntu.com trusty/main Sources [1335 kB]
Get:9 http://archive.ubuntu.com trusty/universe Sources [7926 kB]
Get:10 http://archive.ubuntu.com trusty/multiverse Sources [211 kB]
Get:11 http://archive.ubuntu.com trusty/main amd64 Packages [1743 kB]
Get:12 http://archive.ubuntu.com trusty/multiverse amd64 Packages [169 kB]
Get:13 http://archive.ubuntu.com trusty/universe amd64 Packages [7589 kB]
Fetched 20.8 MB in 53s (389 kB/s)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/main/binary-amd64/Packages  Hash Sum mismatch

E: Some index files failed to download. They have been ignored, or old ones used instead.
The command '/bin/sh -c apt-get update' returned a non-zero code: 100
"

**Comment by bendk** [comment]
I have no idea what the docker quickstart terminal does.  I would recommend inserting that line in the Dockerfile right before the first apt-get install line

**Comment by tuanbs** [comment]
Hi, I open the file Dockerfile and saw the line "RUN apt-get update" in the middle -> Then I put it at the top (after "FROM ubuntu:14.04") and run: bin/dev build -> I got the following error:
"
Sending build context to Docker daemon 404.9 MB
Step 1 : FROM ubuntu:14.04
 ---> c4bea91afef3
Step 2 : RUN apt-get update
 ---> Running in 8cfda380db66
Ign http://archive.ubuntu.com trusty InRelease
Get:1 http://archive.ubuntu.com trusty-updates InRelease [64.4 kB]
Get:2 http://archive.ubuntu.com trusty-security InRelease [64.4 kB]
Hit http://archive.ubuntu.com trusty Release.gpg
Get:3 http://archive.ubuntu.com trusty-updates/main Sources [311 kB]
Get:4 http://archive.ubuntu.com trusty-updates/restricted Sources [5219 B]
Get:5 http://archive.ubuntu.com trusty-updates/universe Sources [183 kB]
Get:6 http://archive.ubuntu.com trusty-updates/main amd64 Packages [836 kB]
Get:7 http://archive.ubuntu.com trusty-updates/restricted amd64 Packages [23.4 kB]
Get:8 http://archive.ubuntu.com trusty-updates/universe amd64 Packages [432 kB]
Get:9 http://archive.ubuntu.com trusty-security/main Sources [129 kB]
Get:10 http://archive.ubuntu.com trusty-security/restricted Sources [3920 B]
Get:11 http://archive.ubuntu.com trusty-security/universe Sources [36.9 kB]
Get:12 http://archive.ubuntu.com trusty-security/main amd64 Packages [501 kB]
Get:13 http://archive.ubuntu.com trusty-security/restricted amd64 Packages [20.2 kB]
Get:14 http://archive.ubuntu.com trusty-security/universe amd64 Packages [159 kB]
Hit http://archive.ubuntu.com trusty Release
Get:15 http://archive.ubuntu.com trusty/main Sources [1335 kB]
Get:16 http://archive.ubuntu.com trusty/restricted Sources [5335 B]
Get:17 http://archive.ubuntu.com trusty/universe Sources [7926 kB]
Get:18 http://archive.ubuntu.com trusty/main amd64 Packages [1743 kB]
Get:19 http://archive.ubuntu.com trusty/restricted amd64 Packages [16.0 kB]
Get:20 http://archive.ubuntu.com trusty/universe amd64 Packages [7589 kB]
Fetched 21.4 MB in 59s (361 kB/s)
W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-updates/main/binary-amd64/Packages  Hash Sum mismatch

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-security/main/source/Sources  Hash Sum mismatch

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-security/main/binary-amd64/Packages  Hash Sum mismatch

W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/trusty-security/universe/binary-amd64/Packages  Hash Sum mismatch

E: Some index files failed to download. They have been ignored, or old ones used instead.
The command '/bin/sh -c apt-get update' returned a non-zero code: 100
"

**Comment by bendk** [comment]
Yeah this seems like an issue with the ubuntu packages and/or the docker base image.  I'm not really sure what to suggest though, maybe try `docker pull ubuntu:14.04`.  Maybe change `ubuntu:14.04` to `ubuntu:15.04` or `ubuntu:15.10`.  

I'm going to close this one because it doesn't seem like an amara issue, sorry I can't be of more help.

**Comment by tuanbs** [comment]
But I think it's part of the Amara issues as it relates to Amara project running. Anyway, thanks for your time. I'll keep searching this issue on Google.

**Comment by tuanbs** [comment]
Hi, I found the cause of this issue. It's because of my internet provider who bans some download urls. I switched to another internet provider and ran step 3: bin/dev build successfully, although there was another issue at this step "Setting up ca-certificates-java" which I solved using "overlay" machine thanks to this topic: https://github.com/docker/docker/issues/18180 -> But now I come up with another issue at step 4: bin/dev up -> The error is below:
"
Creating unisubs20160121_cache_1
Creating unisubs20160121_queue_1
Creating unisubs20160121_db_1
Creating unisubs20160121_worker_1
Creating unisubs20160121_app_1
Attaching to unisubs20160121_worker_1, unisubs20160121_app_1
worker_1 | /usr/bin/python: can't open file 'manage.py': [Errno 2] No such file or directory
unisubs20160121_worker_1 exited with code 2
app_1    | /usr/bin/python: can't open file 'manage.py': [Errno 2] No such file or directory
unisubs20160121_app_1 exited with code 2
"

I guest the command: "bin/dev up" wants to execute file manage.py which does not exist in folder bin/dev. Another weird thing I notice is that after running step 3: "bin/dev build" successfully, there's no new files generated in folder bin/dev. Anyway, I'll open this issue in another thread.
